From 0a8467914ffdefa869addefe51b87eafc125a8d0 Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Tue, 26 Jun 2018 16:32:39 +0200
Subject: [PATCH 3/6] libril: Bounds check s_commands

* We shouldn't be assuming this is always going to be valid

Change-Id: Ibd43133340d86f7c60ead5f7de63c73b05d00f2e
---
 libril/ril.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libril/ril.cpp b/libril/ril.cpp
index ceec055..9b4e28c 100644
--- a/libril/ril.cpp
+++ b/libril/ril.cpp
@@ -233,6 +233,11 @@ addRequestToList(int serial, int slotId, int request) {
 #endif
 #endif
 
+    if (request >= (int)NUM_ELEMS(s_commands)) {
+        RLOGE("Request %s not supported", requestToString(request));
+        return NULL;
+    }
+
     pRI = (RequestInfo *)calloc(1, sizeof(RequestInfo));
     if (pRI == NULL) {
         RLOGE("Memory allocation failed for request %s", requestToString(request));
-- 
2.17.1

